image-upload

```diff
@@ -43,9 +44,8 @@ def upload(dest, source, public):
     if s3.is_key_present(url):
         headers = {s3.ACL: s3.PUBLIC} if public else {}
 
-        # slightly magic: image filenames always end with -{sha256}.qcow2 or -{sha256}.iso
-        assert (source.endswith('.qcow2') or source.endswith('.iso'))
-        hash_value = source.rstrip(".qcow2").rstrip(".iso")[-64:]
+        # slightly magic: we already know the checksum: it's part of the filename
```

Comment:
I think the asserting part did not have to be removed.

Reply:
It didn't, but it's also not important anymore, so I removed it.  The new code will happily deal with any filename, as long as it has a 64 character hex string in it.

Reply:
Indeed. Thanks then!